File: baekjoon_OnlineJudge/Algorithms/DFS_BFS/11724.py
import sys ; input = lambda : sys.stdin.readline().rstrip()

N, M = map(int, input().split())
matrix=[[0]*(N+1) for i in range(N+1)]
for _ in range(M):
    u, v = map(int, input().split())
    matrix[u][v] = matrix[v][u] = 1
visit_list = [0] * (N+1)

count = 0

def dfs(V):
    visit_list[V] = 1
    for i in range(1, N+1):
        if(visit_list[i] == 0 and matrix[V][i]==1):
            dfs(i)


for i in range(1, N+1):
    if not visit_list[i]:
        dfs(i)
        count += 1

print(count)

File: baekjoon_OnlineJudge/Algorithms/DFS_BFS/11724.cpp
#include <iostream>
#define MAX 1001
using namespace std;

int matrix[MAX][MAX];
int N, M;
bool visitList[MAX];

void dfs(int a) {
	visitList[a] = 1;
	for (int i = 1; i <= N; i++) {
		if (!visitList[i] && matrix[a][i]) {
			dfs(i);
		}
	}
}

int main() {
	cin >> N >> M;
	for (int i = 0; i < M; i++) {
		int U, V;
		cin >> U >> V;
		matrix[U][V] = 1;
		matrix[V][U] = 1;
	}
	int count = 0;
	for (int i = 1; i <= N; i++) {
		if (!visitList[i]) {
			dfs(i);
			count += 1;
		}
	}
	cout << count << '\n';
}
